l10n_it_fatturapa_in/migrations/12.0.1.19.0/post-migration.py

```diff
@@ -0,0 +1,38 @@
+# Copyright 2020 Sergio Corato <https://github.com/sergiocorato>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def create_withholding_data_lines(env):
+    # create ftpa_withholding_ids from ftpa_withholding_type and ftpa_withholding_amount
+    openupgrade.logged_query(
+        env.cr, """
+            INSERT INTO withholding_data_line
+            (
+                name,
+                amount,
+                invoice_id,
+                create_uid,
+                create_date,
+                write_date,
+                write_uid
+            )
+            SELECT
+                ai.ftpa_withholding_type,
+                ai.ftpa_withholding_amount,
```

Comment:
@sergiocorato il post migrated script viene eseguito dopo l'aggiornamento del modulo, di conseguenza questo campo `ftpa_withholding_amount`, nel momento dell'esecuzione dello script non è più presente: come fa pertanto a funzionare?

Su un nostro cliente che usa Odoo.sh otteniamo questo errore:

```
10/15/2020 16:00:29ERRORserverFailed to load registry Traceback (most recent call last): File "/home/odoo/src/odoo/odoo/modules/registry.py", line 86, in new odoo.modules.load_modules(registry._db, force_demo, status, update_module) File "/home/odoo/src/odoo/odoo/modules/loading.py", line 417, in load_modules force, status, report, loaded_modules, update_module, models_to_check) File "/home/odoo/src/odoo/odoo/modules/loading.py", line 313, in load_marked_modules perform_checks=perform_checks, models_to_check=models_to_check File "/home/odoo/src/odoo/odoo/modules/loading.py", line 227, in load_module_graph migrations.migrate_module(package, 'post') File "/home/odoo/src/odoo/odoo/modules/migration.py", line 190, in migrate_module migrate(self.cr, installed_version) File "/home/odoo/.local/lib/python3.6/site-packages/openupgradelib/openupgrade.py", line 1796, in wrapped_function if use_env2 else cr, version) File "/home/odoo/src/user/OCA/l10n-italy/l10n_it_fatturapa_in/migrations/12.0.1.19.0/post-migration.py", line 38, in migrate create_withholding_data_lines(env) File "/home/odoo/src/user/OCA/l10n-italy/l10n_it_fatturapa_in/migrations/12.0.1.19.0/post-migration.py", line 30, in create_withholding_data_lines """ File "/home/odoo/.local/lib/python3.6/site-packages/openupgradelib/openupgrade.py", line 1262, in logged_query cr.execute(query, args) File "/home/odoo/src/odoo/odoo/sql_db.py", line 148, in wrapper return f(self, *args, **kwargs) File "/home/odoo/src/odoo/odoo/sql_db.py", line 225, in execute res = self._obj.execute(query, params) psycopg2.ProgrammingError: column ai.ftpa_withholding_amount does not exist LINE 14: ai.ftpa_withholding_amount, ^
```
che mi pare riconducibile a quanto scrivo sopra. Condividi l'analisi o mi sfugge qualcosa?

Reply:
vedi https://github.com/OCA/l10n-italy/pull/1914

Reply:
Cioè chi lo elimina `ftpa_withholding_amount`? Odoo stesso quando rimuovi il campo dal model?

Io aggiornato a `12.0.2.0.0.99.dev2` ma a questo punto non so perchè abbia funzionato

Reply:
> Odoo stesso quando rimuovi il campo dal model?

eh sì

> aggiornato a `12.0.2.0.0.99.dev2` ma a questo punto non so perchè abbia funzionato

cioè hai usato https://github.com/OCA/l10n-italy/pull/1914? funziona perché lo script di migrazione controlla se tale campo esiste prima di effettuare la query.

Reply:
Ho usato https://pypi.org/project/odoo12-addon-l10n-it-fatturapa-in/12.0.2.0.0.99.dev2/

Reply:
> Odoo stesso quando rimuovi il campo dal model?
Ho assunto di sì AFAIK, mi pare logico, @eLBati tu dici di no? cc @sergiocorato

Reply:
In effetti assumevo di no, forse in vecchie versioni era no. Ma magari ora è sì :exploding_head: :grin: 

Sarebbe da provare

Reply:
Si, se esiste la colonna `ftpa_withholding_amount` la copia e la usa, altrimenti fa solo la `ftpa_withholding_type`